### PR TITLE
Redis Session

### PR DESCRIPTION
**Pull request by merrickread**
If `REDIS_SESSION` is true, use `redis-session` for session management, otherwise default to memory store.

`REDIS_SESSION` default is false

### CODE REVIEW

File: src/main-mw.ts [hunk 1]
```diff
@@ -18,7 +19,13 @@ export default function(): Middleware {
     }),
     problem(),
     session({
-      store: 'memory',
+      store: process.env.REDIS_SESSION ? new RedisStore({
+        prefix: 'A12N-session',
```
Comment (mihok) [inline src/main-mw.ts:23]: Might be worthwhile to have this as a env var as well

**Review by mihok — APPROVED** [review]
LGTM

File: src/main-mw.ts [hunk 2]
```diff
@@ -18,7 +19,13 @@ export default function(): Middleware {
...
+        clientOptions: {
+          port: process.env.REDIS_PORT ? parseInt(process.env.REDIS_PORT, 10) : 6379,
+          host: process.env.REDIS_HOST ? process.env.REDIS_HOST : '127.0.0.1',
+        },
+      }) : 'memory',
```
Comment (evert) [inline src/main-mw.ts:28]: Can you remove the `REDIS_SESSION` environment variable, and just use the existence of `REDIS_HOST` as a way to determine we're going to use Redis?

**Review by evert — CHANGES_REQUESTED** [review]
Looks good otherwise!

File: src/main-mw.ts (hunk 2 above)
Comment (merrickread) [reply src/main-mw.ts:28]: yep!

**Review by evert — APPROVED** [review]

File: src/main-mw.ts (hunk 1 above)
Comment (evert) [reply src/main-mw.ts:23]: agreed, could be helpful if multiple environments use the same redis instance. Maybe something like REDIS_PREFIX ?

**Comment by merrickread** [comment]
evert for merging this PR is there anything I need to do regarding versioning?

**Pull request merged**